File: server/src/main/java/moe/yuuta/server/mipush/SendMessageResponse.java
package moe.yuuta.server.mipush;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@SuppressWarnings("unused")
@JsonIgnoreProperties(ignoreUnknown = true)
public class SendMessageResponse {
    public static final String RESULT_OK = "ok";
    public static final String RESULT_ERROR = "error";

    public static final int CODE_SUCCESS = 0;

    @JsonProperty("result")
    private String result;

    @JsonProperty("description")
    private String description;

    @JsonProperty("data")
    private Data data;

    @JsonProperty("code")
    private int code;

    @JsonProperty("info")
    private String info;

    public String getResult() {
        return result;
    }

    public String getDescription() {
        return description;
    }

    public Data getData() {
        return data;
    }

    public int getCode() {
        return code;
    }

    public String getInfo() {
        return info;
    }

    public static class Data {
        private String id;

        public String getId() {
            return id;
        }
    }
}

File: server/src/main/kotlin/moe/yuuta/server/mipush/SendMessageResponse.kt
package moe.yuuta.server.mipush

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@SuppressWarnings("unused")
@JsonIgnoreProperties(ignoreUnknown = true)
data class SendMessageResponse(
        @JsonProperty("result") var result: String,
        @JsonProperty("description") var description: String,
        @JsonProperty("data") var data: SendMessageResponse.Data,
        @JsonProperty("code") var code: Int,
        @JsonProperty("info") var info: String) {
    companion object {
        const val RESULT_OK = "ok"
        const val RESULT_ERROR = "error"

        const val CODE_SUCCESS = 0
    }
    class Data {
        private val id = ""
    }
}
